osquery/tables/system/darwin/cpu_time.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/*
+ *  Inspired from the psutil per_cpu_times function -
+ *  https://github.com/giampaolo/psutil/blob/ec1d35e41c288248818388830f0e4f98536b93e4/psutil/_psutil_osx.c#L739
+ */
+
+#include <mach/mach.h>
+#include <time.h>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+long int ticks_to_usecs(int ticks) {
+  return static_cast<long int>(
+      (static_cast<double>(ticks) / CLOCKS_PER_SEC * 1000000));
+}
+
+QueryData genCpuTime(QueryContext& context) {
+  QueryData results;
+
+  natural_t processor_count;
+  processor_cpu_load_info_data_t* processor_times;
+  mach_port_t host = mach_host_self();
+  mach_msg_type_number_t processor_msg_count;
+
+  kern_return_t ret =
+      host_processor_info(host,
```

Comment:
The first 2 links do 
```
vm_deallocate(mach_task_self(), (vm_address_t)infoArray, infoCount);
``` 
where the `mach_msg_type_number_t` is used the last argument to vm_deallocate. But the last link does 
```
vm_deallocate(mach_task_self(), (vm_address_t)cpuLoad, (vm_size_t)(processorCount * sizeof(*cpuLoad)));
```
where the size of the array is used as the argument. I went with the latter as it makes more sense in terms of `vm_deallocate`s [man page](http://web.mit.edu/darwin/src/modules/xnu/osfmk/man/vm_deallocate.html) which mentions the size as the last argument.